tests/test_client.py

```diff
@@ -2,23 +2,33 @@
 from unittest import TestCase
 
 import pytest
+from unittest import mock
 from alma import ApiModes, Client
+from alma.credentials import ApiKeyCredentials, MerchantIdCredentials, AlmaSessionCredentials
+from alma.request import Request
 from alma.version import __version__ as alma_version
 
 
 class ClientTest(TestCase):
-    def test_client_raises_with_no_api_key_arguments(self):
-        with pytest.raises(
-            TypeError, match=r"__init__\(\) missing 1 required positional argument: 'api_key'"
-        ):
+    def test_client_exposes_api_key_credentials_convenience_method(self):
+        assert Client.with_api_key
+
+    def test_client_exposes_merchant_id_credentials_convenience_method(self):
+        assert Client.with_merchant_id
+
+    def test_client_exposes_alma_session_credentials_convenience_method(self):
+        assert Client.with_alma_session
```

Comment:
We should try that calling this methods actually create the object with the credentials rather than testing if the method exists.